lib/sim.c

```diff
@@ -892,14 +904,18 @@ sim_update_parental_ancestry(sim_t *self, unsigned int parent,
  * Generates parental individuals and inserts them into the parent
  * buffer based on the individuals in the child_buffer. 
  */
+
 static int 
-sim_generate_parents(sim_t *self)
+sim_generate_parents(sim_t *self, int arg_flag)
 {
     int ret = 0;
-    unsigned int j, parent, previous_locus;
+    unsigned int i, j, parent, previous_locus, break_point, gap_no, current_parent;
     individual_t **C = self->child_buffer;
     avl_node_t *node;
     int_map_value_t *locus_mapping;
+    double total_rate, site_1, site_2, coin;
+    gap_no = avl_count(&C[0]->ancestry) - 1;
+    double recomb_rates[gap_no];
```

Comment:
This is dodgy --- allocating anything variable length off the stack can blow up very nastily. A good pattern to follow here is
```c
double recomb_rates = NULL;
recomb_rates = xmalloc(gap_no * sizeof(double));

// Then, later on
out:
    if (recomb_rates != NULL) {
        free(recomb_rates);
    }
```
This involves a malloc for every event, but we can make a buffer for that later if we need to.